### PR TITLE
Add Battle Formations

### PR DESCRIPTION
**Pull request by nhydock**
Addresses #21 by adding a new Node type called Formation.  Extracts out a lot of the spawning logic of battlers into a scene into a factory method.  Formations are used as a form of combat template, and thus also contain meta information such as the kind of environment graphics used when encountering the formation that will also be used when preparing the CombatArena.

Due to how the factory works, also looking into pulling player battlers out of the CombatArena in some way so their stats and presence are persistent across scenes.

### CODE REVIEW

File: godot/combat/CombatArena.tscn [hunk 1]
```diff
@@ -129,114 +96,6 @@ mouse_default_cursor_shape = 0
 size_flags_horizontal = 1
 size_flags_vertical = 1
 alignment = 0
-_sections_unfolded = [ "Grow Direction", "Margin" ]
```
Comment (NathanLovato) [inline godot/combat/CombatArena.tscn]: Can't wait for this option to remove unfolded section from the scene files to be available 😄

**Comment by NathanLovato** [comment]
Looking good so far, thanks a lot 🙂

**Comment by NathanLovato** [comment]
Looking great to me, thanks a lot for your work! Thanks to you, Guilherme, and saalazar, I could spend some time working on art and pimping up the project :)

**Comment by nhydock** [comment]
two Party.gd files got committed accidentally.  The one that's in the root godot folder shouldn't exist

**Comment by NathanLovato** [comment]
Fixed, thanks for the heads-up :)

**Pull request merged**